fortify-issue-manager/src/main/java/com/microfocus/security/automation/fortify/issue/tracker/JiraTrackerClient.java

```diff
@@ -0,0 +1,117 @@
+import okhttp3.OkHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class JiraTrackerClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JiraTrackerClient.class);
+
+    private final static int CONNECTION_TIMEOUT = 30; // seconds
+    private final static int WRITE_TIMEOUT = 600; // seconds
+    private final static int READ_TIMEOUT = 600; // seconds
+
+    private final String apiUrl;
+    private final OkHttpClient client;
+    private final Map<String, String> proxySettings;
+
+    private final String encodedAuth;
+
+    private final static String restApiPath = "/rest/api/2/issue";
+
+    JiraTrackerClient(final BugTrackerSettings bugTrackerSettings) {
+        this.apiUrl = bugTrackerSettings.getApiUrl();
+        this.proxySettings = bugTrackerSettings.getProxySettings();
+
+        client = createClient();
+
+        final String auth = bugTrackerSettings.getUsername() + ":" + bugTrackerSettings.getPassword();
+        encodedAuth = Base64.encodeAsString(auth.getBytes());
+    }
+
+
+    public String performPostRequest(final String payload) throws IOException, BugTrackerException {
+        final HttpUrl apiUrl = HttpUrl.parse(getApiUrl());
+        final String url = apiUrl.newBuilder().addPathSegments(restApiPath).build().toString();
+        LOGGER.debug("Performing request POST {}", url);
+
+        final RequestBody requestBody = RequestBody.create(MediaType.parse("application/json"), payload);
+
+        final Request request = new Request.Builder()
+            .url(url)
+            .addHeader("Authorization", "Basic " + getBasicAuthToken())
+            .post(requestBody)
+            .build();
+
+        final Response response = client.newCall(request).execute();
+        if (!response.isSuccessful()) {
+            throw new BugTrackerException("Failed to create issue for payload : " + payload);
+        }
+
+        // Read the result
+        try (final InputStream responseStream = response.body().byteStream()) {
+            final String responseContent = IOUtils.toString(responseStream, "utf-8");
+            LOGGER.debug("performPostRequest response: {}", responseContent);
+            return responseContent;
+        }
+    }
+
+    private OkHttpClient createClient() {
+        final OkHttpClient.Builder baseClient = new OkHttpClient().newBuilder()
+            .connectTimeout(CONNECTION_TIMEOUT, TimeUnit.SECONDS)
+            .writeTimeout(WRITE_TIMEOUT, TimeUnit.SECONDS)
+            .readTimeout(READ_TIMEOUT, TimeUnit.SECONDS);
+
+        if (!proxySettings.isEmpty()) {
+            final Proxy proxy = new Proxy(
+                Proxy.Type.HTTP,
+                new InetSocketAddress(
+                    proxySettings.get("host"),
+                    Integer.valueOf(proxySettings.get("port"))
+                )
+            );
+            baseClient.proxy(proxy);
+        }
+        return baseClient.build();
+    }
+
+    public String getBasicAuthToken() {
```

Comment:
Use member variable directly instead of this function

Reply:
Done